cmake/modules/GoogleTest.cmake

```diff
@@ -16,15 +16,20 @@ if(WIN32)
     -Dgtest_force_shared_crt=ON
     BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --config $<CONFIG>)
 elseif(APPLE)
+  #FIXME: Temporarily remove -pedantic from CMAKE_CXX_FLAGS to avoid issues with building
+  #       latest version of gtest on MacOS (use of -Wc++17-attribute-extensions causes 
+  #       issues due to [[maybe_unused]] in gtest)
   set(EXTRA_GTEST_OPTS -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT})
+  set(CMAKE_CXX_FLAGS_TEMP_COPY ${CMAKE_CXX_FLAGS})
+  string(REPLACE "-pedantic" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

Comment:
I am wondering if we need to propagate `CMAKE_CXX_COMPILER` to the `CMAKE_ARGS`. Alternatively in `CMAKE_ARGS` we can append `-no-pedantic` to disable the default mode.

Reply:
ping.